_docs/contracts.md

```diff
@@ -113,6 +113,11 @@ Restrictions for libraries in comparison to contracts:
 
 (these might be lifted at a later point)
 
+Common pitfalls for libraries
+
+* The value for `msg.sender` will be that of the contract which is calling the library function.  (If A calls contract B which internally calls library C, then within the function call of library C, `msg.sender` will be the address of contract B).
+* When declaring local variables for structs or arrays, modifications to them will not persist in storage unless they are declared with `storage` as their storage location.  (For example, `var my_struct = self.some_mapping[key]` any changes to `my_struct` would not be persisted in contract storage).
```

Comment:
Would you be ok leaving a small note about it here and linking to the "reference types" section for more detail?  This was something that bit me more than once thus far.